### ISSUE TITLE
In-Game: [HTML5] audio_resume_sound() can crash the game with "Failed to execute 'start' on 'AudioBufferSourceNode'"

### ISSUE THREAD

**Issue by biyectivo** [YoYoGames/GameMaker-HTML5#411]
When trying to pause and then resume a sound, the HTML5 runner crashes with the following error message:

`Unhandled Exception - Uncaught TypeError: Failed to execute 'start' on 'AudioBufferSourceNode': The provided double value is non-finite. in file XXX at line YYY`

I'm attaching a sample barebones project, but the code I'm using is straightforward:

```
if (keyboard_check_pressed(vk_space)) {
	self.music = !self.music;
	if (self.music) {
		if (audio_is_paused(snd_Music))		audio_resume_sound(snd_Music);
		else audio_play_sound(snd_Music, 1, true);
	}
	else {
		if (audio_is_playing(snd_Music))	audio_pause_sound(snd_Music);
	}
}
```

* Confirmed crash on latest public HTML5 runtime (December 15th 2023 commit in the `develop` branch)
* Confirmed crash on latest Beta (2023.1100.0.450 / 2023.1100.0.459)
* Can't trace back to what commit it worked, but I assure you it was working. I reviewed some other old projects and I did have a try/catch block around the resume logic on some, but I also tried using that and it still does not work - it actually does not catch the exception.

Just for kicks, I added a try/catch block **directly on the `Function_Sound.js` file** of the HTML5 runner, on line 425:

Was:
```
this.pbuffersource.start(0, startOffset);
```

After the test:
```
try {
		this.pbuffersource.start(0, startOffset);
	}
	catch (exception) {
		this.pbuffersource.start(0, 0);
	}
```

And this prevents the crash (as `startOffset` seems to be `NaN` and hence the aforementioned error), although it does not _resume_ the sound obviously since I'm using 0 as the new `startOffset` on the `catch` block it _restarts_ it.

Example project:

[audio issue html5.zip](https://github.com/YoYoGames/GameMaker-HTML5/files/13719506/audio.issue.html5.zip)

**Comment by toby-yoyo** [comment]
Found https://github.com/YoYoGames/GameMaker-Bugs/issues/2556 while looking at this.

**Comment by Emc1923** [comment]
Issue still present with the sample attached in  IDE v2024.200.0.454  Runtime v2024.200.0.466 (Press spacebar twice to get the error in the webconsole)

**Comment by toby-yoyo** [comment]
After some investigation, I think the issue still present is in fact https://github.com/YoYoGames/GameMaker-Bugs/issues/4419

Once that issue has been resolved, this one should be able to be verified.

**Comment by toby-yoyo** [comment]
Now that the second issue https://github.com/YoYoGames/GameMaker-Bugs/issues/4419 has been resolved, this one should be fixed.

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.200.0.472  Runtime v2024.200.0.488 when changing the room_width and room_height to numbers. Note that the project attached in the report helped raise the following issue with room_width and room_height in HTML5: https://github.com/YoYoGames/GameMaker-Bugs/issues/4482
